wgpu-core/Cargo.toml

```diff
@@ -83,41 +94,58 @@ glsl = ["naga/glsl-in"]
 ## Enable `ShaderModuleSource::SpirV`
 spirv = ["naga/spv-in", "dep:bytemuck"]
 
+#! ### Other
+# --------------------------------------------------------------------
+
+## Internally count resources and events for debugging purposes. If the counters
+## feature is disabled, the counting infrastructure is removed from the build and
+## the exposed counters always return 0.
+counters = ["wgpu-types/counters"]
+
 ## Implement `Send` and `Sync` on Wasm, but only if atomics are not enabled.
-##
-## WebGL/WebGPU objects can not be shared between threads.
-## However, it can be useful to artificially mark them as `Send` and `Sync`
-## anyways to make it easier to write cross-platform code.
-## This is technically *very* unsafe in a multithreaded environment,
-## but on a wasm binary compiled without atomics we know we are definitely
-## not in a multithreaded environment.
 fragile-send-sync-non-atomic-wasm = [
     "wgpu-hal/fragile-send-sync-non-atomic-wasm",
-    "wgpu-types/fragile-send-sync-non-atomic-wasm",
 ]
 
-#! ### Backends, passed through to wgpu-hal
+#! ### External libraries
 # --------------------------------------------------------------------
+#! The following features facilitate integration with third-party supporting libraries.
 
-## Enable the `metal` backend.
-metal = ["wgpu-hal/metal"]
+## Enable using the `mach-dxcompiler-rs` crate to compile DX12 shaders.
+static-dxc = ["wgpu-hal/static-dxc"]
 
-## Enable the `vulkan` backend.
-vulkan = ["wgpu-hal/vulkan"]
-
-## Enable the `GLES` backend.
-##
-## This is used for all of GLES, OpenGL, and WebGL.
-gles = ["wgpu-hal/gles"]
-
-## Enable the `dx12` backend.
-dx12 = ["wgpu-hal/dx12"]
+#! ### Target Conditional Features
+# --------------------------------------------------------------------
+# Look to wgpu-hal's Cargo.toml for explaination how these features and the wgpu-core
+# platform crates collude to provide platform-specific behavior.
+
+## DX12 backend
+dx12 = ["wgpu-core-platform-windows-linux-android-emscripten/dx12"]
+## Metal backend
+metal = ["wgpu-core-platform-apple/metal"]
+## Vulkan backend, only available on Windows, Linux, Android
+vulkan = ["wgpu-core-platform-windows-linux-android-emscripten/vulkan"]
```

Comment:
Not that anyone _actually_ cares about this I reckon, but I wonder if that will drag in Vulkan dependencies on emscripten? Since that's wasm, it should be blocked off on the hal level again, right? But it will drag in some of the "general" dependencies.

If so that's worth documenting that this is a cost we accept.

Reply:
Ah I see - ash won't be pulled in, but the general dependencies will be. We should just have a separate crate for emscripten. That's too high a cost imo.

Reply:
Split into its own crate.